File: 2021_09_29/Pascal/4.pas
var ans : longint;
	x1, d : longint;

function isPrime(n : longint) : Boolean;
var i : longint;
Begin
	for i := 2 to round(sqrt(n)) do
		if (n mod i = 0) then
		begin
			isPrime := False;
			Exit;
		end;
	isPrime := True;
End;

BEGIN
	ans := 0;
	for x1 := 1000 to 9999 do
	begin
		if not isPrime(x1) then
			continue;
		for d := 1 to (10000 - x1) div 3 do
			if isPrime(x1+d) and isPrime(x1+2*d) and isPrime(x1+3*d) then
				Inc(ans);
	end;
	Writeln(ans);
END.

File: 2021_09_29/C++/4.cpp
#include <iostream>

bool isPrime(int n) {
	for (int i = 2; i * i <= n; i++)
		if (n % i == 0)
			return false;
	return true;
}

int main() {
	int ans = 0;
	for (int x1 = 1000; x1 < 10000; x1++) {
		if (!isPrime(x1))
			continue;
		for (int d = 1; d <= (10000 - x1) / 3; d++)
			if (isPrime(x1 + d) && isPrime(x1 + 2 * d) && isPrime(x1 + 3 * d))
				ans++;
	}
	std::cout << ans << std::endl;
	return 0;
}
